### DISCUSSION TITLE
NIC recommendation for large image

### DISCUSSION THREAD

**Discussion by tfordon** [ipxe/ipxe#570] [Q&A]
Does anybody have a NIC they would recommend for deploying large IPXE images?

I would like to deploy an IPXE image with embedded certificates and a script.  The image has gotten pretty large (85k) and I am concerned it won't fit on most NICs.  I heard most Intel NICs only have a 64k expansion ROM.

Thank you for your time,
Tom

(edited: fixed filesize)

**Comment by mcb30** [comment]
> I would like to deploy an IPXE image with embedded certificates and a script. The image has gotten pretty large (~700k)

Is this a typo?  Did you mean ~70k?

Michael

**Reply by tfordon** [reply]
Sorry, i was looking at the wrong file.  Thank you for spotting my blunder!  I am currently at 85k.

I haven't tried reducing the file size with undefs yet, so I might be able to get it below 64k.  But if there is a NIC that has a larger expansion ROM it would be helpful.

**Comment by 9001** [comment]
Looks like all i350 cards have the same flash size, 524288 bytes -- tried a t4v1 and a t2v2, 536 KiB was too big, 509 KiB was aight

Should mention one funky thing with i350 is that it has multiple ROM banks, one for each port, and the BIOS can choose to read any of the ROMs at boot -- even ones that are disabled by intel's bootutil. So remember to write your image to all the ROMs

**Comment by stappersg** [comment]
@tfordon please share with the iPXE community (the part that uses Github.com) your solution or workaround.
